src/span.js

```diff
@@ -222,14 +168,11 @@ export default class Span {
     }
 
     /**
-     * Indicates that the unit of work represented by the span is complete or
-     * has otherwise been terminated.
-     *
-     * All Span objects must have finish() called on them before they are
-     * reported to the backend implementation.
+     * Sets the end timestamp and finalizes Span state.
      *
-     * Once `finish()` is called on a Span object, the behavior of all methods
-     * on the object is considered undefined.
+     * With the exception of calls to Span.context() (which are always allowed),
```

Comment:
(talked about this offline, but TL;DR the `context()` after `finish()` is here for a reason, and it does make it harder to use a pool)